React/Modules/RCTDevLoadingView.m

```diff
@@ -57,59 +64,64 @@ - (void)setBridge:(RCTBridge *)bridge
 
 - (void)showWithURL:(NSURL *)URL
 {
-  dispatch_async(dispatch_get_main_queue(), ^{
-
-    _showDate = [NSDate date];
-    if (!_window && !RCTRunningInTestEnvironment()) {
-      CGFloat screenWidth = [UIScreen mainScreen].bounds.size.width;
-      _window = [[UIWindow alloc] initWithFrame:CGRectMake(0, 0, screenWidth, 22)];
-      _window.backgroundColor = [UIColor blackColor];
-      _window.windowLevel = UIWindowLevelStatusBar + 1;
-
-      _label = [[UILabel alloc] initWithFrame:_window.bounds];
-      _label.font = [UIFont systemFontOfSize:12.0];
-      _label.textColor = [UIColor grayColor];
-      _label.textAlignment = NSTextAlignmentCenter;
-
-      [_window addSubview:_label];
-      [_window makeKeyAndVisible];
-    }
-
-    NSString *source;
-    if (URL.fileURL) {
-      source = @"pre-bundled file";
-    } else {
-      source = [NSString stringWithFormat:@"%@:%@", URL.host, URL.port];
-    }
-
-    _label.text = [NSString stringWithFormat:@"Loading from %@...", source];
-    _window.hidden = NO;
-
-  });
+  if (isEnabled) {
```

Comment:
I prefer not to have multiple exits of a given method like that, as you end up eventually having hard to follow logic. I will make this change shortly since you've asked for it, but do you have a style guide I can reference for things like this?

> On Aug 26, 2015, at 6:32 PM, James Ide <notifications@github.com> wrote:
> 
> In React/Modules/RCTDevLoadingView.m:
> 
> > -      [_window addSubview:_label];
> > -      [_window makeKeyAndVisible];
> > -    }
> > -
> > -    NSString *source;
> > -    if (URL.fileURL) {
> > -      source = @"pre-bundled file";
> > -    } else {
> > -      source = [NSString stringWithFormat:@"%@:%@", URL.host, URL.port];
> > -    }
> > -
> > -    _label.text = [NSString stringWithFormat:@"Loading from %@...", source];
> > -    _window.hidden = NO;
> > -
> > -  });
> > +  if (isEnabled) {
> Just bail out & return early if !isEnabled to avoid all this indentation.
> 
> ―
> Reply to this email directly or view it on GitHub.
>